File: articles/food-water-and-land-use-futures/julia/food_water_land_resilience.jl
# Julia standard-library example for food-water-land resilience scoring.

profiles = Dict(
    "Regenerative Transition" => [0.72, 0.76, 0.82, 0.78, 0.76, 0.42, 0.38, 0.80, 0.78],
    "High-Yield Degradation" => [0.86, 0.44, 0.32, 0.30, 0.50, 0.58, 0.62, 0.42, 0.46],
    "Watershed Restoration" => [0.68, 0.82, 0.78, 0.84, 0.72, 0.46, 0.44, 0.76, 0.80]
)

function fwl_resilience(v)
    # v = production water soil biodiversity governance climate market justice livelihoods
    0.13*v[1] + 0.16*v[2] + 0.15*v[3] + 0.14*v[4] + 0.14*v[5] -
    0.12*v[6] - 0.08*v[7] + 0.14*v[8] + 0.12*v[9]
end

println("profile,food_water_land_resilience")
for (name, values) in sort(collect(profiles))
    println("$(name),$(round(fwl_resilience(values), digits=4))")
end

File: articles/food-water-and-land-use-futures/cpp/food_water_land_resilience.cpp
#include <iostream>

double fwl_resilience(double production, double water, double soil, double biodiversity, double governance, double climate, double market, double justice, double livelihoods) {
    return 0.13 * production + 0.16 * water + 0.15 * soil + 0.14 * biodiversity
        + 0.14 * governance - 0.12 * climate - 0.08 * market
        + 0.14 * justice + 0.12 * livelihoods;
}

int main() {
    std::cout << "Regenerative Transition resilience="
              << fwl_resilience(0.72, 0.76, 0.82, 0.78, 0.76, 0.42, 0.38, 0.80, 0.78) << "\n";
    return 0;
}
